app/src/androidTest/java/com/duckduckgo/app/statistics/api/StatisticsRequesterJsonTest.kt

```diff
@@ -0,0 +1,280 @@
+/*
+ * Copyright (c) 2018 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.statistics.api
+
+import android.support.test.InstrumentationRegistry
+import com.duckduckgo.app.FileUtilities.loadText
+import com.duckduckgo.app.InstantSchedulersRule
+import com.duckduckgo.app.global.AppUrl.ParamKey
+import com.duckduckgo.app.statistics.Variant
+import com.duckduckgo.app.statistics.VariantManager
+import com.duckduckgo.app.statistics.model.Atb
+import com.duckduckgo.app.statistics.store.StatisticsDataStore
+import com.duckduckgo.app.statistics.store.StatisticsSharedPreferences
+import com.nhaarman.mockito_kotlin.mock
+import com.nhaarman.mockito_kotlin.whenever
+import com.squareup.moshi.Moshi
+import okhttp3.OkHttpClient
+import okhttp3.mockwebserver.MockResponse
+import okhttp3.mockwebserver.MockWebServer
+import org.junit.After
+import org.junit.Assert.*
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import retrofit2.Retrofit
+import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
+import retrofit2.converter.moshi.MoshiConverterFactory
+
+
+class StatisticsRequesterJsonTest {
```

Comment:
I did wonder about that, myself.

I figured it would be good to put the new tests in the separate place since they were focused on the JSON handling using mock web server, and the original tests were purely using mocks.

They could be put in the same suite, but I figured it would make for a more confusing setup as some tests would be using mock web server and some would just be using `Mockito` mocks. Totally doable, but I hoped this way would be nicer.

Reply:
Your reasoning makes, sense 👍